tests/test_setup_command.py

```diff
@@ -16,5 +39,134 @@ def test_name_and_version(project) -> None:
     project.setup_cfg(setup_cfg)
     project.setup_py()
     result = project.run()
-    assert result.returncode == 0
-    assert result.stdout == "running pyproject\n" + pyproject_toml
+    check_result(result, pyproject_toml)
+
+
+# install_requires tests, we use made-up module names here.
+
+
+def test_install_requires(project) -> None:
+    """
+    Test the install_requires is passed through if given.
+    """
+    setup_cfg = """\
+[metadata]
+name = test-project
+version = 0.0.1
+
+[options]
+install_requires =
+        dependency1
+        dependency2>=1.23
+        dependency3<4.56
+"""
+    pyproject_toml = """\
+[build-system]
+requires = ["setuptools"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "test-project"
+version = "0.0.1"
+dependencies = ["dependency1", "dependency2>=1.23", "dependency3<4.56"]
+"""
+    project.setup_cfg(setup_cfg)
+    project.setup_py()
+    result = project.run()
+    check_result(result, pyproject_toml)
+
+
+# setup_requires tests: we have to use real dependencies that actually are
+# installed, otherwise the tests will fail.
```

Comment:
I hope at some point we can figure out a way around this, but nothing to do about it for now.